### PR TITLE
remove scratchpad, define roles

### PR DESCRIPTION
**Pull request by mattvarghese**
This pull request 
* removes the extraneous notion of scratchpad, which accurately belongs inside the EHR abstraction. Just like the CDS Hooks specification being able to fully work without defining any such arbitrary scratchpad, this specifications is also now able to fully work without defining such a scratchpad.
* cleans up the activity-catalog to correctly distinguish what is an EHR activity that SMART Web Messaging can navigate to, as different from a user workflow point inside an activity that would trigger a CDS Hook.
* creates a consistent response message structure. This now also includes a `messageType` for the response message which usually is `app.response`.
* clarifies that the app initiates the handshake. If we need an EHR initiated handshake (use-case doesn't exist today), the semantics of that is different, and we can formulate that as a different interaction at that point in the future.
* defines three roles `ui`, `app`, and `fhir` proxy. Stays deliberately silent on additional roles.
* change authorization scope to include full message type. Previously text said scope was at `message group` level, but examples showed scope was at full message type. Fixed this discrepancy by going with the model in the examples, which feels more persuasive.
* correct JSON RFC and a bunch of such minor errors.

### CODE REVIEW

**Comment by mattvarghese** [comment]
I'm also sticking with the perspective that `ui.*` is the appropriate prefix for `create`, `update`, and `delete` messages, since for example we're asking to create an unsigned order in the EHR UI as opposed to one in the background when doing so through the REST-ful endpoint.

**Comment by mattvarghese** [comment]
A list of issues that I believe are addressed through this PR
* https://jira.hl7.org/browse/FHIR-31052
* https://jira.hl7.org/browse/FHIR-29368
* https://jira.hl7.org/browse/FHIR-29349
* (may be more - update as necessary)

File: fsh/ig-data/input/pagecontent/index.md [hunk 1]
```diff
@@ -6,9 +6,9 @@
 *[API]: Application Programming Interface
 *[CDS]: Clinical Decision Support
 *[CPOE]: Computerized Physician Order Entry
-*[CRUD]: Copy Read Update Delete
+*[CRUD]: Create Read Update Delete
```
Comment (barabo) [inline fsh/ig-data/input/pagecontent/index.md:9]: Wow!  Um, thanks for catching that.  Yikes

Comment (mattvarghese) [reply fsh/ig-data/input/pagecontent/index.md:9]: Ha ha, don't feel bad though - I'm sure all of us have made such mistakes!

**Comment by mattvarghese** [comment]
Based on a conversation today with @jmandel, @barabo, and @GinoCanessa, I should add this note as a PSA about what is the benefit / advantage of SMART Web Messaging over just contacting the restful API.
* It is NOT that we have the ability to CRUD draft / work-in-progress resources. For example, FHIR allows doing a MedicationRequest CREATE on the FHIR REST-ful endpoint where the status of the MedicationRequest is draft. Doing so will cause the newly created MedicationRequest to end up in the same place in the EHR that all other orders being created end up in.. So it's not any kind of scratchpad access that is the benefit - **this example demonstrates that the REST-ful API is intended to have the same access**.
* Rather, EHRs usually have UI/Client, and database. Doing the above CREATE operation on the REST-ful FHIR API will cause the draft MedicationRequest to be created in the server portion of the EHR - what is variously called scratchpad. However, unlike normally in EHRs where orders are created by the user on the client and sync to this scratchpad, this order got created "from left field" on the EHR, so that the **client is not aware of this order**. That IS **the most relevant problem**. This can potentially result in the end user never realizing that there is an order, and the order going to the void, instead of getting signed by the end user.

So **what SMART Web Messaging really needs is a _UI mediated_ way to CRUD FHIR resources on the EHR**. Please note the emphasis on "UI mediated".

It's not about giving any special access to special locations or anything like that - suggesting such ideas necessitate tearing open the abstraction of the black box that is the EHR and is bad practice for a spec that defines an interface because that is impinging on implementation.

**Comment by jmandel** [comment]
Thanks for this feedback -- we've incorporated some of these suggestions through separate PRs as part of our ballot reconciliation process; others we've taken a different path or deferred.

**Pull request closed without merging**